cmd/podman/common/create.go

```diff
@@ -15,6 +17,40 @@ const sizeWithUnitFormat = "(format: `<number>[<unit>]`, where unit = b (bytes),
 
 var containerConfig = registry.PodmanConfig()
 
+// MapOptions takes the Container and Pod Create options, assigning the matching values back to podCreate for the purpose of the libpod API
+func MapOptions(containerCreate entities.ContainerCreateOptions, podCreate entities.PodCreateOptions) (entities.PodCreateOptions, entities.ContainerCreateOptions) {
```

Comment:
I understand that you like this because you don't have to manually set `createOptions.Cpus = infraOptions.CPUS` but I really don't like the use of reflect here. Reflect is slow and using it correctly is complicated. Your function can contain serious bugs, it could easily panic.

Reply:
Yes reflect can have its issues but in thus scenario we use it so carefully that we will only be comparing and assigning values that we know are identical in type. Also the setting of the variables has been set up in a very safe way where we check the type for each available option before setting it. We don't just use `.Set`  this function makes pod creation much more easy to read and understand in the long run

Reply:
Is there a reason we can't use straight JSON (encode the pod struct, decode into a container struct), given your addition of the appropriate tags to both structs? Substantially less error-prone, *MUCH* faster, and we use it everywhere already.

Reply:
Sure @mheon I can give that a shot. Didn't think of that